packages/edit-navigation/src/components/sidebar/delete-menu.js

```diff
@@ -2,31 +2,40 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, PanelBody } from '@wordpress/components';
+import {
+	Button,
+	PanelBody,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
 
 export default function DeleteMenu( { onDeleteMenu, isMenuBeingDeleted } ) {
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+
 	return (
 		<PanelBody>
-			<Button
-				className="edit-navigation-inspector-additions__delete-menu-button"
-				variant="secondary"
-				isDestructive
-				isBusy={ isMenuBeingDeleted }
-				onClick={ () => {
-					if (
-						// eslint-disable-next-line no-alert
-						window.confirm(
-							__(
-								'Are you sure you want to delete this navigation? This action cannot be undone.'
-							)
-						)
-					) {
-						onDeleteMenu();
-					}
-				} }
-			>
-				{ __( 'Delete menu' ) }
-			</Button>
+			<>
+				<Button
+					className="edit-navigation-inspector-additions__delete-menu-button"
+					variant="secondary"
+					isDestructive
+					isBusy={ isMenuBeingDeleted }
+					onClick={ () => {
+						setShowConfirmDialog( true );
+					} }
```

Comment:
Small optimisation: you could look into memoizing this callback so that the function doesn't get recreated every time the `DeleteMenu` component renders (see [related comment](https://github.com/WordPress/gutenberg/pull/37491#discussion_r780804949))

Reply:
It isn't necessarily a performance optimization though, [sometimes it might get worse](https://kentcdodds.com/blog/usememo-and-usecallback). I'd prefer to keep it simple until we audit it.

Reply:
Sure, we can keep it as-is unless an optimisation is needed

Reply:
@ciampo I know several of your other comments on this PR (and the other's you've reviewed for me) also touched on memoizing callbacks. They're all similar callbacks, does the approach of holding off until optimization is needed apply to those recommendations as well?

Reply:
Let's hold off for now, it's always something that we can add later on — it will also allow you to focus on other aspects of the feedback first :)